src/System.Net.Security/src/System/Net/Unix/SslConnectionInfo.cs

```diff
@@ -74,12 +80,19 @@ private static ExchangeAlgorithmType MapExchangeAlgorithmType(long mac)
             {
                 case 1:
                     return ExchangeAlgorithmType.RsaKeyX;
-                case 2 | 32:
-                    return ExchangeAlgorithmType.RsaSign;
-                case 128:
+                case 2:
+                case 4:
+                case 8:
                     return ExchangeAlgorithmType.DiffieHellman;
+                case 16:
+                case 32:
+                case 64:
+                case 128:
+                case 256:
+                case 512:
+                case 1024:
+                    return (ExchangeAlgorithmType) mac;
```

Comment:
There are few algorithms that Windows/Unix can support but are not listed in ENUMs. (ie. ECDH, SHA256 and many more) 
In windows also some number is returned representing that value , Same thing we followed here ,